apps/examples/net_local/net_local_main.c

```diff
@@ -0,0 +1,154 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <strings.h>
+#include <unistd.h>
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <sched.h>
+#include <ctype.h>
+#include <pthread.h>
+
+#include "net_local.h"
+
+#define MAXLINE 80
+
+
+int net_local_server_start(void)
+{
+	int pid = task_create("net_local_server", SCHED_PRIORITY_DEFAULT, 4096, net_local_server_main, (FAR char *const *)NULL);
+	if (pid < 0) {
+		printf("net_local_server start failed, err = %d\n", pid);
+	}
+	return pid;
+}
+
+int net_local_client_start(void)
+{
+	int pid = task_create("net_local_client", SCHED_PRIORITY_DEFAULT, 4096, net_local_client_main, (FAR char *const *)NULL);
+	if (pid < 0) {
+		printf("net_local_client start failed, err = %d\n", pid);
+	}
+	return pid;
+}
+
+int client_talk(int sockfd, const char *func, const char *str)
+{
+	int n;
+	char buf[MAXLINE + 1];
+	printf("[%s] send: %s\n", func, str);
+	n = write(sockfd, (const void *)str, strlen(str));
+	if (n < 0) {
+		perror("write");
+		return -1;
+	}
+
+	n = read(sockfd, buf, MAXLINE);
+	if (n < 0) {
+		perror("read");
+		return -1;
+	} else {
+		buf[n] = '\0';
+		printf("[%s] recv: %s\n", func, buf);
+		return strncasecmp(buf, str, n);
+	}
+}
+
+void server_talk(int sockfd, const char *func)
+{
+	int i, n;
+	char buf[MAXLINE + 1];
+	while (1) {
+		n = read(sockfd, buf, MAXLINE);
+		if (n < 0) {
+			perror("read");
+			break;
+		} else if (n == 0) {
+			break;
+		}
+
+		buf[n] = '\0';
+		printf("[%s] receive: %s\n", func, buf);
+		for (i = 0; i < n; i++) {
+			buf[i] = toupper(buf[i]);
+		}
+		printf("[%s] respond: %s\n", func, buf);
+		write(sockfd, buf, n);
+	}
+}
+
+static void *socketpair_server(void *param)
+{
+	int fd = (int)param;
+	server_talk(fd, __FUNCTION__);
+	close(fd);
+	return NULL;
+}
+
+int sockpair_testcase1(void)
+{
+	int ret = -1;
+	int fds[2];
+
+	if (socketpair(AF_UNIX, SOCK_STREAM, 0, fds) == -1) {
```

Comment:
Could you let me know the location of `socketpair` function defintion and implementation?